compiler/src/dotty/tools/repl/Rendering.scala

```diff
@@ -64,11 +64,21 @@ private[repl] class Rendering(parentClassLoader: Option[ClassLoader] = None) {
       myClassLoader
     }
 
+  /** Used to elide output in replStringOf.
+   * TODO: Perhaps implement setting scala.repl.maxprintstring as in Scala 2, but
+   * then this bug will surface, so perhaps better not?
+   * https://github.com/scala/bug/issues/12337
+   */
+  private[repl] def truncate(str: String): String =
+    if str.length > MaxStringElements then str.take(MaxStringElements - 3) + "..."
```

Comment:
We should drop 1 on the right if the last character `isUnicodeIdentifierStart` to not cut a Unicode character in half.

Reply:
You mean something like:
```
private[repl] def truncate(str: String): String =
  def adjust(s: String): String = 
    if s.nonEmpty && s.last.isUnicodeIdentifierStart then s.dropRight(1) 
    else s

  if str.length > MaxStringElements then adjust(str.take(MaxStringElements - 3)) + "..."
  else str
```
?
(I don't know so much about how Unicode code points work and I have been trying to read up on https://home.unicode.org/ but just realized that it is a mouthful...)

Reply:
If that works, yes. Maybe you have to use `java.lang.Character.isUnicodeIdentifierStart`.

Reply:
Actually might be a bit harder than that. We also need to check if the last one `isUnicodeIdentifierPart` then the next one that we dropped should not be `isUnicodeIdentifierPart`.

Maybe we should open an issue and let someone more knowledgeable about those detail help.

Reply:
As a note for myself; more to learn about code points here: https://www.unicode.org/faq/utf_bom.html

Reply:
I think it is this one to use: https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#codePointBefore-int-

Reply:
Or rather `offsetByCodePoints` and `codePointCount` as in
```
def trunc(s: String, n: Int) = 
  val n2 = s.codePointCount(0, s.length)
  s.substring(0, if n2 > n then s.offsetByCodePoints(0, n) else s.length)
```

Reply:
I think I found a solution now:
```
  private[repl] def truncate(str: String): String = {
    val ncp = str.codePointCount(0, str.length) // to not cut inside code point
    if ncp > MaxStringElements && ncp > 3 then
      str.substring(0, str.offsetByCodePoints(0, MaxStringElements - 3)) + "..."
    else str
  }
```